arches/app/media/js/viewmodels/reference-select.js

```diff
@@ -0,0 +1,158 @@
+define([
+    'jquery',
+    'knockout',
+    'knockout-mapping',
+    'arches',
+    'viewmodels/widget',
+], function($, ko, koMapping, arches, WidgetViewModel) {
+    var NAME_LOOKUP = {};
+    var ReferenceSelectViewModel = function(params) {
+        var self = this;
+
+        params.configKeys = ['placeholder'];
+        this.multiple = !!ko.unwrap(params.node.config.multiValue);
+        this.displayName = ko.observable('');
+        this.selectionValue = ko.observable([]); // formatted version of this.value that select2 can use
+        this.activeLanguage = arches.activeLanguage;
+
+        WidgetViewModel.apply(this, [params]);
+
+        this.getPrefLabel = function(labels){
+            return koMapping.toJS(labels)?.find(label => label.language===arches.activeLanguage && label.valuetype === 'prefLabel').value;
+        }; 
+
+        this.flattenItems = function(items, flatList, uniqueItems){
+            //TODO Children are duplicated coming from the concept_list endpoint. We can remove 'uniqueItems' once that's fixed
+            items.forEach(item=>{
+                if (uniqueItems.includes(item.uri) === false) {
+                    uniqueItems.push(item.uri);
+                    flatList.push(item);
+                    if (item.children.length) {
+                        self.flattenItems(item.children, flatList, uniqueItems);
+                    }
+                }
+            });
+            flatList.forEach(subItem=>delete subItem.children); //Prevent select2 from symbolizing parent as grouping reference
+            return flatList;
+          };
+          
+        this.displayValue = ko.computed(function() {
+            const val = self.value();
+            let name = '';
+            if (val) {
+                name = val.map(item=>self.getPrefLabel(item.labels)).join(", ");
+            }
+            return val ? name : null;
+        });
+
+        this.selectionValue.subscribe(val => {
+            if (val) {
+                if (!(val instanceof Array)) { val = [val]; }
+                const tileReady = val.map(uri => {
+                    return {
+                        "uri": uri,
+                        "listid": NAME_LOOKUP[uri]["listid"],
+                        "labels": NAME_LOOKUP[uri].labels
+                    };
+                });
+                self.value(tileReady);
+            }
+        });
+
+        this.select2Config = {
+            value: self.selectionValue,
+            clickBubble: true,
+            multiple: this.multiple,
+            closeOnSelect: true,
+            placeholder: self.placeholder,
```

Comment:
Maybe this is a future card/not important, but we don't have any validation on invalid placeholders.